src/read.c

```diff
@@ -863,15 +864,31 @@ static avifResult avifDecoderItemMaxExtent(const avifDecoderItem * item, const a
     uint64_t maxOffset = 0;
     for (uint32_t extentIter = 0; extentIter < item->extents.count; ++extentIter) {
         avifExtent * extent = &item->extents.extent[extentIter];
-        const size_t usedExtentSize = (extent->size < remainingBytes) ? extent->size : remainingBytes;
+
+        // Make local copies of extent->offset and extent->size as they might need to be adjusted
+        // due to the sample's offset.
+        uint64_t startOffset = extent->offset;
+        uint64_t extentSize = extent->size;
+        if (remainingOffset) {
+            if (remainingOffset >= extentSize) {
+                remainingOffset -= extentSize;
+                continue;
+            } else {
+                startOffset += remainingOffset;
```

Comment:
We may need to check for overflow before the addition:

```
        if (remainingOffset> UINT64_MAX - startOffset) {
            return AVIF_RESULT_BMFF_PARSE_FAILED;
        }
```

This check can be omitted if we know extent->offset + extent->size won't overflow.